File: laramie-administracion-webapi/src/domain/entities/expediente.ts
export default class Expediente {

    id: number;
	matricula: string;
	ejercicio: string;
	numero: string;
	letra: string;
	idProvincia: number;
	idTipoExpediente: number;
	subnumero: string;
	idTemaExpediente: number;
	referenciaExpediente: string;
	fechaCreacion: Date;

	constructor(
        id: number = 0,
		matricula: string = "",
		ejercicio: string = "",
		numero: string = "",
		letra: string = "",
		idProvincia: number = 0,
		idTipoExpediente: number = 0,
		subnumero: string = "",
		idTemaExpediente: number = 0,
		referenciaExpediente: string = "",
		fechaCreacion: Date = null
	)
	{
        this.id = id;
		this.matricula = matricula;
		this.ejercicio = ejercicio;
		this.numero = numero;
		this.letra = letra;
		this.idProvincia = idProvincia;
		this.idTipoExpediente = idTipoExpediente;
		this.subnumero = subnumero;
		this.idTemaExpediente = idTemaExpediente;
		this.referenciaExpediente = referenciaExpediente;
		this.fechaCreacion = fechaCreacion;
	}

	setFromObject = (row) =>
	{
        this.id = row.id ?? 0;
		this.matricula = row.matricula ?? "";
		this.ejercicio = row.ejercicio ?? "";
		this.numero = row.numero ?? "";
		this.letra = row.letra ?? "";
		this.idProvincia = row.idProvincia ?? 0;
		this.idTipoExpediente = row.idTipoExpediente ?? 0;
		this.subnumero = row.subnumero ?? "";
		this.idTemaExpediente = row.idTemaExpediente ?? 0;
		this.referenciaExpediente = row.referenciaExpediente ?? "";
		this.fechaCreacion = row.fechaCreacion ?? null;
	}

}

File: modules/administracion/models/expediente.py
from sqlalchemy import Column, BigInteger, String, Integer, DateTime, func

import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", ".."))
from shared.database import Base


class Expediente(Base):
    __tablename__ = "administracion_expedientes"

    id = Column(BigInteger, primary_key=True, autoincrement=True)
    matricula = Column(String(50), nullable=True)
    ejercicio = Column(Integer, nullable=False)
    numero = Column(Integer, nullable=False)
    letra = Column(String(10), nullable=True)
    id_tipo_expediente = Column(BigInteger, nullable=True)
    subnumero = Column(Integer, nullable=True)
    referencia_expediente = Column(String(500), nullable=True)
    fecha_creacion = Column(DateTime(timezone=True), server_default=func.now())
